src/serialbox-c/CMakeLists.txt

```diff
@@ -32,29 +32,37 @@ if(SERIALBOX_ENABLE_C)
   # Compile serialboxC
   set(SOURCES
     Archive.cpp
+    Archive.h
     Array.cpp
+    Array.h
     ConfigOptions.cpp
+    ConfigOptions.h
     ErrorHandling.cpp
+    ErrorHandling.h
     FieldMetainfo.cpp
+    FieldMetainfo.h
     Logging.cpp
+    Logging.h
     Metainfo.cpp
+    Metainfo.h
     Savepoint.cpp
+    Savepoint.h
     Serializer.cpp
+    Serializer.h
     FortranWrapper.cpp
+    FortranWrapper.h
   )
 
   set(EXTERNAL_LIBRARIES ${Boost_LIBRARIES} ${SERIALBOX_EXTERNAL_LIBRARIES})
 
-  set(SYMLINKS ${CMAKE_BINARY_DIR}/src/serialbox-python/serialbox
-               ${CMAKE_SOURCE_DIR}/src/serialbox-python/serialbox)
+  yoda_create_library(

```

Comment:
This is another interesting case, similar to the `yoda_add_unittest`, but here instead, we create a library. 
Do you remember the discussion on the propagation of the include directories of the dependent targets? Well, here it also happens, all include directories required by SerialboxCoreStatic will be propagated to the compilation of our target here. 
But additionally, since we are creating a library, that can be used as a dependency of another target later on, it is also time that we describe the properties of this target, like include directories, with two purposes: we need to specify them for the compilation of our own sources, and at the same time, we want to specify the include directories that should be propagated to other targets that will depend on this one. 
For a full API description, see the documentation of yoda: 
https://github.com/cosunae/yoda/blob/create_library/cmake/yodaCreateLibrary.cmake

Let focus on the semantic of passing include directories: 
there are keyword like `[PUBLIC|PRIVATE|INTERFACE]_[BUILD|INSTALL]_INCLUDES`
* _PUBLIC_ means that the include directory is needed to compile our target and also need to be propagated to targets to depend on ours. 
* _PRIVATE_ means that the include directory is only needed to compile our target
* _INTERFACE_ means that the include directory will be needed only to propagate to target that depend on ours. 
On the second word: 
* _BUILD_ refers to paths of the build structure
* _INSTALL_ refers to paths of the install structure

As we can see here, we need to specify two different paths, for the build and install structure. This is because, common practice in many projects, the headers of serialbox are found under `serialbox/src/` and the corresponding structure of subprojects. But they will all be installed under `${CMAKE_INSTALL_PREFIX}/include`